src/runtime/components/forms/Form.vue

```diff
@@ -132,9 +132,15 @@ export default defineComponent({
         if (props.validateOn?.includes('submit')) {
           await validate()
         }
+        let submitterName: string | null = null
+        if (event.submitter) {
+          const element = event.submitter as HTMLButtonElement | HTMLInputElement
+          submitterName = element.name || null
+        }
         const submitEvent: FormSubmitEvent<any> = {
           ...event,
```

Comment:
The issue came from this, the event's attributes were not passed on properly. I submitted another PR to properly expose the source event's attributes including the submitter: https://github.com/nuxt/ui/pull/2012